src/compat.jl

```diff
@@ -0,0 +1,505 @@
+        end
+    elseif e.head === :module
+        @assert nargs === 3
+        if !e.args[1]
+            st_flags |= JS.BARE_MODULE_FLAG
+        end
+        child_exprs = [e.args[2], e.args[3]]
+    elseif e.head === :do
+        # Expr:
+        # (do (call f args...) (-> (tuple lam_args...) (block ...)))
+        # SyntaxTree:
+        # (call f args... (do (tuple lam_args...) (block ...)))
+        callargs = collect_expr_parameters(e.args[1], 2)
+        fname = string(callargs[1])
+        if fname[1] === '@'
+            st_k = K"macrocall"
+            callargs[1] = Expr(:MacroName, callargs[1])
+        else
+            st_k = K"call"
+        end
+        child_exprs = [callargs..., Expr(:do_lambda, e.args[2].args...)]
+    elseif e.head === :let
+        if nargs >= 1 && !(e.args[1] isa Expr && e.args[1].head === :block)
+            child_exprs[1] = Expr(:block, e.args[1])
+        end
+    elseif e.head === :struct
+        e.args[1] && (st_flags |= JS.MUTABLE_FLAG)
+        child_exprs = child_exprs[2:end]
+        # TODO handle docstrings after refactor
+    elseif (e.head === :using || e.head === :import)
+        _expr_replace!(e,
+                       (e)->(e isa Expr && e.head === :.),
+                       (e)->(e.head = :importpath))
+    elseif e.head === :kw
+        st_k = K"="
+    elseif e.head in (:local, :global) && nargs > 1
+        # Possible normalization
+        # child_exprs = [Expr(:tuple, child_exprs...)]
+    elseif e.head === :error
+        # Zero-child errors from parsing are leaf nodes.  We could change this
+        # upstream for consistency.
+        if nargs === 0
+            child_exprs = nothing
+            st_attrs[:value] = JS.ErrorVal()
+            st_flags |= JS.TRIVIA_FLAG
+        end
+    end
+
+    #---------------------------------------------------------------------------
+    # The following heads are not emitted from parsing, but old macros could
+    # produce these and they would historically be accepted by flisp lowering.
+    if e.head === Symbol("latestworld-if-toplevel")
+        st_k = K"latestworld_if_toplevel"
+    elseif e.head === Symbol("hygienic-scope")
+        st_k = K"hygienic_scope"
+    elseif e.head === :meta
+        # Messy and undocumented.  Only sometimes we want a K"meta".
+        @assert e.args[1] isa Symbol
+        if e.args[1] === :nospecialize
+            if nargs > 2
+                st_k = K"block"
+                # Kick the can down the road (should only be simple atoms?)
+                child_exprs = map(c->Expr(:meta, :nospecialize, c), child_exprs[2:end])
+            else
+                st_id, src = _insert_convert_expr(e.args[2], graph, src)
+                setmeta!(SyntaxTree(graph, st_id); nospecialize=true)
+                return st_id, src
+            end
+        else
+            @assert nargs === 1
+            child_exprs[1] = Expr(:sym_not_identifier, e.args[1])
```

Comment:
`K"Symbol"` is like `QuoteNode(:some_sym)` so we could go with `quoted_symbol`  (Maybe `K"Symbol"` would better be renamed at some point for clarity)

```suggestion
            child_exprs[1] = Expr(:quoted_symbol, e.args[1])
```